addons/udes_stock/tests/test_stock_reservation.py

```diff
@@ -0,0 +1,74 @@
+"""
+Tests for StockPicking.reserve_stock.
+
+The method itself cannot be tested because it contains a commit statement, but
+we can test related methods.
+"""
+from .common import BaseUDES
+
+
+class FindUnreservableMovesTestCase(BaseUDES):
+    """Test cases for StockPicking.NAME_HERE."""
+
+    # u_reserve_batches
+    # reserve batch picking off: pick for unavailable product is not reserved
+    # reserve batch picking off: pick for available product is reserved
+    # reserve batch picking on: multiple picks 1 unavailable: batch not reserved
+    # reserve batch picking on: multiple picks 0 unavailable: batch reserved
+    # u_handle_partials, u_handle_partial_lines
+    # u_num_reservable_pickings
+    # skip states
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.Move = cls.env["stock.move"]
+
+        cls.create_quant(cls.apple.id, cls.test_location_01.id, 10)
+
+    def test_reserves_stock_for_picking_if_available(self):
+        self.picking_type_pick.u_handle_partials = False
+        products_info = [{"product": self.apple, "qty": 10}]
+        picking = self.create_picking(self.picking_type_pick, products_info, confirm=True)
+
+        unreservable_moves = picking._find_unfulfillable_moves()
+
+        self.assertEqual(unreservable_moves, self.Move.browse())
```

Comment:
Suggestion: Could just do `assertFalse` and avoid needing the `Move` class instance. Although happy for to keep it more explicit.